domain/src/main/kotlin/no/nav/su/se/bakover/domain/beregning/SlåSammenEkvivalenteMånedsberegningerTilBeregningsperioder.kt

```diff
@@ -27,63 +28,154 @@ import no.nav.su.se.bakover.domain.beregning.fradrag.Fradrag
  *     a     b     a   b   c
  *   |---|-------|---|---|---|
  */
-internal data class SlåSammenEkvivalenteMånedsberegningerTilBeregningsperioder(
-    private val månedsberegninger: List<Månedsberegning>
-) {
-    val beregningsperioder: List<EkvivalenteMånedsberegninger> = mutableListOf<MutableList<Månedsberegning>>().apply {
-        månedsberegninger.sorterMånedsberegninger().forEach { månedsberegning ->
-            when {
-                this.isEmpty() -> this.add(mutableListOf(månedsberegning))
-                this.last().sisteMånedsberegningErLikOgTilstøtende(månedsberegning) -> this.last().add(månedsberegning)
-                else -> this.add(mutableListOf(månedsberegning))
+
+internal sealed class SlåSammenEkvivalenteMånedsberegningerTilBeregningsperioder {
+    data class Utbetaling(
```

Comment:
Jeg tenker at verken utbetalingsdomenet eller brevdomenet hører hjemme her. Denne kunne vel ha fått lov til å vært som den var, også kunne vi kalt den fra en UtbetalingslinjeMapper som masserer resultatet fra denne klassen videre. Og evt. noe tilsvarende for brev, dersom den og må ta høyde for uføregraden.

Reply:
En løsning kunne kanskje vært at denne algoritmen forble som den var, så kunne heller utbetalingene "lagt på" uføregraden (siden det er der vi har behovet). Kunne f.eks laget en tidslinje av uføregrunnlag og fisket ut det aktuelle for hver distinkte beregningsperiode? Det som da evt ville gjenstått er at utbetalingen må håndtere tilfeller hvor uføregraden varierer innenfor samme "ekvivalente" beregningsperiode. Type, beløp er 5000 for en periode på 2 måneder, men hver av disse månedene har hhv 50 og 100 i uføregrad.

Reply:
Dette er nå trukket ut til utbetalingsdomenet. Denna fila skal være igjen as is.

Reply:
👍